cpp/CMakeLists.txt

```diff
@@ -233,37 +223,48 @@ set(CMAKE_INSTALL_DEFAULT_COMPONENT_NAME cuspatial)
 
 install(TARGETS cuspatial
         DESTINATION lib
-        EXPORT cuspatial-targets)
+        EXPORT cuspatial-exports)
 
 install(DIRECTORY ${CUSPATIAL_SOURCE_DIR}/include/cuspatial
         DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 
-include(CMakePackageConfigHelpers)
-
-configure_package_config_file(cmake/cuspatial-config.cmake.in "${CUSPATIAL_BINARY_DIR}/cmake/cuspatial-config.cmake"
-                              INSTALL_DESTINATION "${INSTALL_CONFIGDIR}")
-
-write_basic_package_version_file("${CUSPATIAL_BINARY_DIR}/cmake/cuspatial-config-version.cmake"
-                                 COMPATIBILITY SameMinorVersion)
-
 install(FILES   "${CUSPATIAL_BINARY_DIR}/cmake/cuspatial-config.cmake"
                 "${CUSPATIAL_BINARY_DIR}/cmake/cuspatial-config-version.cmake"
     DESTINATION "${INSTALL_CONFIGDIR}")
 
-install(EXPORT  cuspatial-targets
-    FILE        cuspatial-targets.cmake
-    NAMESPACE   cuspatial::
-    DESTINATION "${INSTALL_CONFIGDIR}")
+set(doc_string
+    [=[
+Provide targets for the cuSpatial library.
 
-################################################################################################
-# - build export -------------------------------------------------------------------------------
+cuSpatial is a GPU-accelerated library for spatial data management and analytics.
+
+Imported Targets
+^^^^^^^^^^^^^^^^
+
+If cuspatial is found, this module defines the following IMPORTED GLOBAL
+targets:
+
+ cuspatial::cuspatial             - The main cuspatial library.
+    ]=]
+)
+
+rapids_export(
+  INSTALL cuspatial
+  EXPORT_SET cuspatial-exports
+  GLOBAL_TARGETS cuspatial
+  NAMESPACE cuspatial::
+  DOCUMENTATION doc_string
+)
 
-configure_package_config_file(cmake/cuspatial-build-config.cmake.in ${CUSPATIAL_BINARY_DIR}/cuspatial-config.cmake
-                              INSTALL_DESTINATION ${CUSPATIAL_BINARY_DIR})
 
-write_basic_package_version_file(${CUSPATIAL_BINARY_DIR}/cuspatial-config-version.cmake
-                                 COMPATIBILITY SameMinorVersion)
+################################################################################################
+# - build export -------------------------------------------------------------------------------
 
-export(EXPORT cuspatial-targets
-    FILE ${CUSPATIAL_BINARY_DIR}/cuspatial-targets.cmake
-    NAMESPACE   cuspatial::)
+# TODO: I'm not sure how these two exports should differ here.
```

Comment:
They should differ by the `BUILD` keyword only. This tells rapids-cmake to place the information in the build directory so it can be used to find cuspatial. Without that you would need to first install cuspatial to find it via CMake's `find_package`.